### ISSUE TITLE
Adversarial Loss not trained against stagnant partner?

### ISSUE THREAD

**Issue by yhu9** [ai-tor/PyTorch-CycleGAN#12]
I was under the impression that the optimizer and the generator should be training against stagnant counterparts but the step function on the generators are called before the gradients are calculated for the discriminator. Is this standard practice?

**Comment by FloCF** [comment]
For e.g. DCGAN or pix2pix people usually start with discriminator training with detached generated fakes and then optimize Generator. Not sure why it is the other way round here.
Dont know if that helps :sweat_smile: ...
